providers/component-protocol/components/baselist/mode.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package baselist
+
+import "github.com/erda-project/erda-infra/providers/component-protocol/cptype"
+
+type (
+	// BaseList includes basic list of items.
+	BaseList struct {
+		Type       string                                   `json:"type,omitempty"`
+		Data       ListData                                 `json:"data,omitempty"`
+		Operations map[cptype.OperationKey]cptype.Operation `json:"operations,omitempty"`
+		UserIDs    []string                                 `json:"userIDs,omitempty"`
+	}
+
+	// ListData .
+	ListData struct {
+		List []Item `json:"list"`
+	}
+
+	// Item minimum unit of list
+	Item struct {
+		// uniq id of the item, e.g: appID, projID, ...
+		ID    string `json:"id,omitempty"`
+		Title string `json:"title,omitempty"`
+		// logo link of title
+		Logo          string                                   `json:"logo,omitempty"`
+		Labels        []ItemLabel                              `json:"labels,omitempty"`
+		Description   string                                   `json:"description,omitempty"`
+		BackgroundImg string                                   `json:"backgroundImg,omitempty"`
+		Operations    map[cptype.OperationKey]cptype.Operation `json:"operations,omitempty"`
```

Comment:
I think you should add `fieldList` and enum all supported field type, such as KVList, card, hover and others.